src/pkgcheck/checks/python.py

```diff
@@ -268,6 +274,49 @@ def desc(self):
         updates = ', '.join(self.updates)
         return f'PYTHON_COMPAT update{s} available: {updates}'
 
+class PythonOptionalCheck(Check):
+    """Check Python ebuilds for missing optional dependencies.
+
+    The problematic case for us is DISTUTILS_OPTIONAL,
+    DISTUTILS_USE_PEP517 != no, but no ${DISTUTILS_DEPS} anywhere.
+    """
+    _restricted_source = (sources.RestrictionRepoSource, (
+            packages.PackageRestriction('inherited', values.ContainmentMatch2('distutils-r1')),
+        )
+    )
+    _source = (sources.EbuildParseRepoSource, (), (('source', _restricted_source),))
+
+    known_results = frozenset([
+        PythonMissingOptionalDeps
+    ])
+
+    def feed(self, item):
+        has_distutils_optional = None
+        has_distutils_pep517_non_no = None
+
+        for var_node, _ in bash.var_assign_query.captures(item.tree.root_node):
+            var_name = item.node_str(var_node.child_by_field_name('name'))
+
+            if var_name == "DISTUTILS_OPTIONAL":
+                has_distutils_optional = True
+
+            if "DISTUTILS_DEPS" in item.node_str(var_node.parent):
+                # If they're referencing the eclass' dependency variable,
+                # there's nothing for us to do anyway.
+                return
+
+            if var_name == "DISTUTILS_USE_PEP517" and not has_distutils_pep517_non_no:
+                var_val = item.node_str(var_node.children[-1])
+
+                # For DISTUTILS_USE_PEP517=standalone, the eclass doesn't
+                # provide ${DISTUTILS_DEPS}.
```

Comment:
If I understand the comment correctly, this isn't true currently - as we have `gpep517` dep for `standalone`. Only in `=no` case we have `bdep=''`

Reply:
Sorry, yes, that's outdated from when I changed my mind about implementation

Reply:
Ah, I'd pushed wrong branch :)